modules/c++/six.sicd/source/Utilities.cpp

```diff
@@ -130,6 +130,166 @@ Utilities::getProjectionModel(const ComplexData* data,
     }
 }
 
+six::NITFReadControl six::sicd::Utilities::getLoadedNITFReadControl(
+        const std::string& sicdPathname,
+        const std::vector<std::string>& schemaPaths)
+{
+    six::XMLControlRegistry xmlRegistry;
+    xmlRegistry.addCreator(six::DataType::COMPLEX,
+            new six::XMLControlCreatorT<
+            six::sicd::ComplexXMLControl>());
+
+    six::NITFReadControl reader;
+    reader.setXMLControlRegistry(&xmlRegistry);
```

Comment:
We could statically add all of Six's XMLControlCreator to the singleton XMLControlFactory, that way we'd never have to explicitly touch the XMLRegistry when we want to load a file. The weird thing (that is already there, I guess) is that six::Utilities::parseData() needs to know about SICD and SIDD, and I guess it just doesn't know about CPHD. We could get around that by adding a virtual `canParse(string)` function to each XMLControlCreator (or somewhere convenient). That way XMLControlRegistry can ask all of its creators if they know how to parse particular XML and return the right one without parseData knowing about what it's loading.